### PR TITLE
Add slot table in PS.Parameters

### PR DESCRIPTION
**Pull request by mhaoli**
Part of #1306 , step 1.

### CODE REVIEW

File: elasticdl/python/ps/embedding_table.py [hunk 1]
```diff
@@ -15,10 +15,15 @@ class EmbeddingTable(object):
     dictionary, the corresponding value will be initialized.
     """
 
-    def __init__(self, name, dim=None, initializer=None):
+    def __init__(self, name, dim=None, initializer=None, is_slot=False):
```
Comment (terrytangyuan) [inline elasticdl/python/ps/embedding_table.py:18]: Add some docs for each arg

File: elasticdl/python/ps/parameters.py [hunk 2]
```diff
@@ -99,6 +103,8 @@ def init_from_model_pb(self, model_pb):
             self._init_embedding_params(embeddings_pb)
             self.version = model_pb.version
             self.init_status = True
+            return True
```
Comment (terrytangyuan) [inline elasticdl/python/ps/parameters.py:117]: Combine the two return statements to one return `self.init_status` instead?

Comment (mhaoli) [reply elasticdl/python/ps/parameters.py:117]: > Combine the two return statements to one return `self.init_status` instead?

The return value does not equal to `init_status`, it means whether `init_from_model_pb` accepts this model pb. I will add a docs here.

File: elasticdl/python/ps/parameters.py [hunk 3]
```diff
@@ -110,3 +116,22 @@ def _init_non_embedding_params(self, tensors_pb):
...
         for pb in embeddings_pb:
             self.embedding_params[pb.name] = create_embedding_table(pb)
+
+    def has_embedding_params(self):
+        return len(self.embedding_params) > 0
+
+    def create_slot_params(self, slot_names, init_values):
+        embed_layer_names = list(self.embedding_params.keys())
+        for layer_name in embed_layer_names:
+            for slot_name in slot_names:
```
Comment (QiJune) [inline elasticdl/python/ps/parameters.py:137]: One question, are the slots of all the embedding layers the same?

File: elasticdl/python/ps/embedding_table.py (hunk 1 above)
Comment (mhaoli) [reply elasticdl/python/ps/embedding_table.py:18]: Done

File: elasticdl/python/ps/parameters.py (hunk 3 above)
Comment (mhaoli) [reply elasticdl/python/ps/parameters.py:137]: Yes, it is decided by the optimizer type. Adam creates slots "m" and "v" for each variable, and  Adagrad creates slots "accumulator" for each variable. The user-defined arguments of initial values are also in optimizer.

**Review by QiJune — APPROVED** [review]
Excellent!

**Pull request merged**